src/benchmarklib/tpch/tpch_query_generator.cpp

```diff
@@ -25,9 +30,84 @@ void TPCHQueryGenerator::_generate_names() {
   }
 }
 
+std::string TPCHQueryGenerator::get_preparation_queries() const {
+  if (!_use_prepared_statements) return "";
+
+  std::stringstream sql;
+  for (auto query_id = QueryID{0}; query_id < 22; ++query_id) {
+    if (query_id + 1 == 15) {
+      // We cannot prepare query 15, because the SELECT relies on a view that is generated in the first step. We'll have
+      // to manually build this query once we start randomizing the parameters.
+      continue;
+    }
+
+    auto query_template = std::string{tpch_queries.find(query_id + 1)->second};
+
+    // Escape single quotes
+    boost::replace_all(query_template, "'", "''");
+
+    sql << "PREPARE TPCH" << (query_id + 1) << " FROM '" << query_template << "';\n";
+  }
+  return sql.str();
+}
+
 std::string TPCHQueryGenerator::build_query(const QueryID query_id) {
   DebugAssert(query_id < 22, "There are only 22 TPC-H queries");
-  return tpch_queries.find(query_id + 1)->second;
+
+  if (query_id == 14) {
+    // Generating TPC-H Query 15 by hand
+    auto query_15 = std::string{tpch_queries.find(15)->second};
+
+    static auto view_id = 0;
+    boost::replace_all(query_15, std::string("revenueview"), std::string("revenue") + std::to_string(view_id++));
+    return query_15;
+  }
+
+  // Stores how the parameters (the ? in the query) should be replaced
```

Comment:
I'll add a comment, but for those minimal tests, we do not use the demo values defined in the standard.

Reply:
// Stores how the parameters (the ? in the query) should be replaced. These values are examples for the queries. Most
  // of them use the validation parameters given in the TPC-H specification for the respective query. A few are
  // modified so that we get results even for a small scale factor.